### ISSUE TITLE
OpenGL sky breaks when playing at more than 90 fov :/

### ISSUE THREAD

**Issue by El-Juancho** [coelckers/prboom-plus#385]
90 FOV:
![image](https://user-images.githubusercontent.com/70743149/131232861-6ea95a4a-e9ec-4142-812a-eb4dddd0550e.png)

95 FOV:
![image](https://user-images.githubusercontent.com/70743149/131232893-1c247352-f4e6-48b6-8b5f-a133f103109e.png)
This is eviternity MBF complevel.

**Comment by JadingTsunami** [comment]
It's expected, there is no perfect way to display skies at >90 FOV.

**Comment by El-Juancho** [comment]
> It's expected, there is no perfect way to display skies at >90 FOV.
> 
> EDIT: Note most skies will not look as bad as Eviternity, it's something of a special case.

oh :( ,should i close this issue?

**Comment by JadingTsunami** [comment]
> > It's expected, there is no perfect way to display skies at >90 FOV.
> > EDIT: Note most skies will not look as bad as Eviternity, it's something of a special case.
> 
> oh :( ,should i close this issue?

Oh, a last thing -- you could try setting gl_skymode to 3 in your config file. You may find the sky more visually acceptable that way. I also edited in a few other options in my previous response.

As for now I don't plan more work on the GL sky renderer. I suggest skymode 3 as a good balanced choice to fix issues such as this one.

**Comment by fabiangreffrath** [comment]
> I suggest skymode 3 as a good balanced choice to fix issues such as this one.

Should this become the new default probably?

**Comment by JadingTsunami** [comment]
> > I suggest skymode 3 as a good balanced choice to fix issues such as this one.
> 
> Should this become the new default probably?

Possibly? I considered it, but there are tradeoffs because right now, the OpenGL renderer matches vanilla as close as possible with freelook disabled (the sky is rendered as a flat scaled texture). After enabling freelook, the sky is rendered as a cylinder.

Skymode 3 will always render the sky as a cylinder, which will be visually nicer with freelook disabled, wide FOV, and non-standard sky types, but doesn't look like the vanilla rendering method otherwise.

**Comment by fabiangreffrath** [comment]
So, does it make sense to change the condition in `M_ChangeMouseLook()` to read 
```
gl_drawskys = ((movement_mouselook || render_fov > FOV90) ? skytype_skydome : skytype_standard);
```
here
https://github.com/coelckers/prboom-plus/blob/5b5bfa6ffcf656c9fcffc74d2d1c33201d27a74b/prboom2/src/e6y.c#L521-L526

**Comment by Kappa971** [comment]
From what I understand, does this make sure that by setting a fov > 90, gl_skymode 3 is automatically set?
With the render software the sky is displayed differently than in gl_skymode 3, is this the right solution?

**Comment by JadingTsunami** [comment]
I'm contemplating just modifying the FOV rendering for flat skies to make it looks more or less the same at all FOVs.

This is not the "right" way to do it but seems to be the behavior that players expect.

**Comment by JadingTsunami** [comment]
Proposing to fix this issue via PR #418.

**Comment by fabiangreffrath** [comment]
https://github.com/coelckers/prboom-plus/commit/9e2c5543cfe9ba97a2b83ce73c2bf1471279d41a
